code/datums/world_topic.dm

```diff
@@ -102,20 +102,25 @@
 	// We can't add the timer without the timer ID, but we can't get the timer ID without the timer!
 	// To solve this, we just use a list that we mutate later.
 	var/list/data = list("input" = input)
-	var/timer_id = addtimer(CALLBACK(src, .proc/receive_cross_comms_message, data), CROSS_SECTOR_CANCEL_TIME, TIMER_STOPPABLE)
+	// Did we have to pass the soft filter on our origin server? Passed as a boolean value.
+	var/soft_filter_passed = input["is_filtered"]
+	var/timer_id = addtimer(CALLBACK(src, .proc/receive_cross_comms_message, data), soft_filter_passed ? EXTENDED_CROSS_SECTOR_CANCEL_TIME : CROSS_SECTOR_CANCEL_TIME, TIMER_STOPPABLE)
 	data["timer_id"] = timer_id
 
 	LAZYADD(timers, timer_id)
 
-	to_chat(
-		GLOB.admins,
-		span_adminnotice( \
-			"<b color='orange'>CROSS-SECTOR MESSAGE (INCOMING):</b> [input["sender_ckey"]] (from [input["source"]]) is about to send \
-			the following message (will autoapprove in [DisplayTimeText(CROSS_SECTOR_CANCEL_TIME)]): \
-			<b><a href='?src=[REF(src)];reject_cross_comms_message=[timer_id]'>REJECT</a></b><br> \
-			[html_encode(input["message"])]" \
-		)
-	)
+	var/extended_time_display = DisplayTimeText(EXTENDED_CROSS_SECTOR_CANCEL_TIME)
+	var/normal_time_display = DisplayTimeText(CROSS_SECTOR_CANCEL_TIME)
+
+	var/message = "<b color='orange'>CROSS-SECTOR MESSAGE (INCOMING):</b> [input["sender_ckey"]] (from [input["source"]]) is about to send \
+			the following message (will autoapprove in [soft_filter_passed ? "[extended_time_display]" : "[normal_time_display]"]): \
+			<b><a href='?src=[REF(src)];reject_cross_comms_message=[timer_id]'>REJECT</a></b><br><br>\
+			[html_encode(input["message"])]"
+
+	if(soft_filter_passed)
```

Comment:
I was seeing this message even when the cross-server message was "hi"

Reply:
i hope hi isn't on the soft filter

i'll look into that though, thanks for catching it.